lib/rules/at-rule-descriptor-no-unknown/__tests__/index.mjs

```diff
@@ -0,0 +1,106 @@
+import rule from '../index.mjs';
+const { messages, ruleName } = rule;
+
+testRule({
+	ruleName,
+	config: true,
+
+	accept: [
+		{
+			code: '@counter-style foo { system: cyclic; symbols: url(gold-medal.svg); suffix: " "; }',
+		},
+		{
+			code: '@property --foo { syntax: "<integer>"; inherits: false; initial-value: 0; }',
+		},
+		// TODO: The following test can be supported once mdn/data, is updated and the css-tree is updated.
+		// see: https://github.com/mdn/data/blob/6b4cae7c0aa96c199a982f2b6efeda7ce67eb515/css/at-rules.json
+		// {
+		// 	code: '@view-transition { navigation: auto; }',
+		// },
+		// {
+		// 	code: '@container (width > 400px) and (height > 400px) { a { width: 50%; } }',
+		// },
+		// {
+		// 	code: '@container not (width > 400px) { a { width: 50%; } }',
+		// },
+		// {
+		// 	code: '@container summary (min-width > 400px) { @container (min-width: 800px) { a { width: 50%; } } }',
+		// },
+	],
+
+	reject: [
+		{
+			code: '@counter-style foo { bar: 0; }',
+			message: messages.rejected('@counter-style', 'bar'),
+			line: 1,
+			column: 22,
+			endLine: 1,
+			endColumn: 29,
+		},
+		{
+			code: '@property --foo { bar: 0; }',
+			message: messages.rejected('@property', 'bar'),
+			line: 1,
+			column: 19,
+			endLine: 1,
+			endColumn: 26,
+		},
+		// TODO: The following test can be supported once mdn/data, is updated and the css-tree is updated.
+		// see: https://github.com/mdn/data/blob/6b4cae7c0aa96c199a982f2b6efeda7ce67eb515/css/at-rules.json
+		// {
+		// 	code: stripIndent`
+		// 		@container (width > 400px) and (height > 400px) {
+		// 			a {
+		// 				foo: 0;
+		// 			}
+		// 		}
+		// 	`,
+		// 	message: messages.rejected('@container', 'foo'),
+		// 	line: 3,
+		// 	column: 3,
+		// 	endLine: 3,
+		// 	endColumn: 10,
+		// },
+		// {
+		// 	code: stripIndent`
+		// 		@container not (width > 400px) {
+		// 			a {
+		// 				foo: 0;
+		// 			}
+		// 		}
+		// 	`,
+		// 	message: messages.rejected('@container', 'foo'),
+		// 	line: 3,
+		// 	column: 3,
+		// 	endLine: 3,
+		// 	endColumn: 10,
+		// },
+		// {
+		// 	code: stripIndent`
+		// 		@container summary (min-width > 400px) {
+		// 			@container (min-width: 800px) {
+		// 				a {
+		// 					foo: 0;
+		// 				}
+		// 			}
+		// 		}
+		// 	`,
+		// 	warnings: [
+		// 		{
+		// 			message: messages.rejected('@container', 'foo'),
+		// 			line: 4,
+		// 			column: 4,
+		// 			endLine: 4,
+		// 			endColumn: 11,
+		// 		},
+		// 		{
+		// 			message: messages.rejected('@container', 'foo'),
+		// 			line: 4,
+		// 			column: 4,
+		// 			endLine: 4,
+		// 			endColumn: 11,
+		// 		},
+		// 	],
+		// },
```

Comment:
```suggestion
```

These are properties within style rules.

Reply:
Sorry, I still don’t understand. I thought this was a test for `@container` [descriptors](https://drafts.csswg.org/css-conditional-5/#container-descriptor-table). Is this actually testing properties inside `@container` style rules instead?

Reply:
> I thought this was a test for @container [descriptors](https://drafts.csswg.org/css-conditional-5/#container-descriptor-table).

That's interesting. It's the first time I've seen "descriptors" used to describe these. Elsewhere in the spec, they are called [container features](https://drafts.csswg.org/css-conditional-5/#container-features).

Shall we limit this rule to checking if "[descriptor declarations](https://drafts.csswg.org/css-syntax/#declaration)" are known for an at-rule? If so, we can delete these tests.

Reply:
Ah, I see! The "descriptors" for `@container` are things like the `width` in this:

```css
@container (width > 600px) {}
```

I misunderstood. Thanks for the detail.